src/main/java/io/airlift/slice/Slice.java

```diff
@@ -498,10 +490,7 @@ public void getShorts(int index, short[] destination)
      */
     public void getShorts(int index, short[] destination, int destinationIndex, int length)
     {
-        checkFromIndexSize(index, length * Short.BYTES, length());
-        checkFromIndexSize(destinationIndex, length, destination.length);
-
-        copyFromBase(index, destination, ARRAY_SHORT_BASE_OFFSET + ((long) destinationIndex * Short.BYTES), length * Short.BYTES);
+        MemorySegment.copy(MemorySegment.ofArray(base), index, MemorySegment.ofArray(destination), (long) destinationIndex * Short.BYTES, (long) length * Short.BYTES);
```

Comment:
Nit: can you format all of these on multiple lines, like the `getBytes` method above?